### ISSUE TITLE
Lint/MissingCopEnableDirective is disabled by rubocop:disable all

### ISSUE THREAD

**Issue by matthewhively** [rubocop/rubocop#14102]
## Expected behavior

A missing magic comment `rubocop:enable all` should be reported

## Actual behavior

No issues are reported.

## Steps to reproduce the problem

test_file.rb
```
def my_function
  return 'foobar'
end
```
1) Run `bundle exec rubocop test_file.rb` and observe no reported issues
2) Edit file to include `# rubocop:disable all` at the top of the file
3) Run `bundle exec rubocop test_file.rb` and observe no reported issues (bugged behavior)
4) Run `bundle exec rubocop --ignore-disable-comments test_file.rb` to ignore the magic comments and observe the following is now reported:
```
Offenses:

test_file.rb:1:1: W: Lint/MissingCopEnableDirective: Re-enable Migration/DepartmentName cop with # rubocop:enable after disabling it.
# rubocop:disable all

1 file inspected, 1 offense detected
```
5) Edit file to append 
```
# rubocop:enabel all  -- note the typo
def another_function
  return 'barfoo'
end
```
6) Run `bundle exec rubocop test_file.rb` and observe no issue is reported

------

I would argue that in the first case I've described where I'm disabling an entire file, is working as I intend, because it is effectively ignoring the entire file. But the way it is doing it, would have side-effects.
Since rubocop has NO way of knowing this is intentional and not meant to only be a sub-section of the file, I believe the `Lint/MissingCopEnableDirective` should not be included within `rubocop:disable all`.

Similar issue for reference: https://github.com/rubocop/rubocop/issues/5933

## RuboCop version

```
1.75.2 (using Parser 3.3.8.0, rubocop-ast 1.44.1, analyzing as Ruby 2.7, running on ruby 2.7.8) [arm64-darwin24]
  - rubocop-performance 1.25.0
  - rubocop-rails 2.31.0
```

**Comment by bbatsov** [comment]
Closing — this is by design. `rubocop:disable all` disables all cops including `Lint/MissingCopEnableDirective`. If you need to keep that cop active, disable specific cops or departments instead of using `all`.
